Framework/DataObjects/src/TimeSplitter.cpp

```diff
@@ -37,6 +37,33 @@ TimeSplitter::TimeSplitter(const DateAndTime &start, const DateAndTime &stop) {
   clearAndReplace(start, stop, DEFAULT_VALUE);
 }
 
+/**
+ * Note: The amount of X values in input MatrixWorkspace must be 1 larger than the amount of Y values.
+ * There are NO undefined split regions here.
+ **/
+TimeSplitter::TimeSplitter(const Mantid::API::MatrixWorkspace_sptr &ws, const DateAndTime &offset) {
+  if (ws->getNumberHistograms() != 1) {
+    throw std::runtime_error("MatrixWorkspace can only have 1 histogram when constructing TimeSplitter.");
+  }
+
+  const auto X = ws->binEdges(0);
+  const auto &Y = ws->y(0);
+  if (std::any_of(X.begin(), X.end(), [](double i) { return static_cast<int>(i) < 0; })) {
+    throw std::runtime_error("All X values in MatrixWorkspace must be >= 0 to construct TimeSplitter.");
+  }
+  if (X.size() != Y.size() + 1) {
+    throw std::runtime_error(
+        "Size of x values must be one more than size of y values to construct TimeSplitter from MatrixWorkspace.");
+  }
+
+  for (size_t i = 1; i < X.size(); i++) {
+    auto timeStart = Types::Core::DateAndTime(static_cast<int64_t>((X[i - 1])), offset.totalNanoseconds());
+    auto timeEnd = Types::Core::DateAndTime(static_cast<int64_t>(X[i]), offset.totalNanoseconds());
+    auto index = static_cast<int>(Y[i - 1]);
+    this->addROI(timeStart, timeEnd, index);
```

Comment:
@gecage952 , Pete previously requested (it was on your PR for `SplittersWorkspace`) that we add a warning before adding an ROI, saying that a time interval might not make it to the ROI collection.  I looked at how you did it originally and slightly modified it in my code, for example:
```
 // if this row's time interval intersects an ROI already in the splitter, no separate ROI will be created in the
    // splitter, hence the warning.
    if (valueAtTime(timeStart) != IGNORE_VALUE || valueAtTime(timeStop) != IGNORE_VALUE) {
      g_log.warning() << "Workspace row " << ii << " may be overwritten in conversion to TimeSplitter: " << '\n';
    }

    addROI(timeStart, timeStop, target_ws_index);

```
Now I am going to make it even more detailed, because I've realized that the target workspace index might be IGNORE_VALUE, in which case the warning just won't make sense.